### PR TITLE
Make controller and infrastructure for booth. and add to context with a one to one relation

### PR DESCRIPTION
**Pull request by Husso1997**

### CODE REVIEW

File: BazarRestAPI/Controllers/BoothsController.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            }
+        }
+
+        // POST: api/Booths - Create booth
+        [HttpPost]
+        public ActionResult<Booth> Post([FromBody] Booth booth)
+        {
+            try
+            {
+                return _service.Create(booth);
```
Comment (asbamse) [inline BazarRestAPI/Controllers/BoothsController.cs]: Missing Ok()

File: BazarRestAPI/Controllers/BoothsController.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        }
+
+        // PUT: api/Booths/5 - Update booth
+        [HttpPut("{id}")]
+        public ActionResult<Booth> Put(int id, [FromBody] Booth booth)
+        {
+            try
+            {
+                booth.Id = id;
+                return _service.Update(booth);
```
Comment (asbamse) [inline BazarRestAPI/Controllers/BoothsController.cs]: Missing Ok()

File: BazarRestAPI/Controllers/BoothsController.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            }
+        }
+
+        // DELETE: api/ApiWithActions/5 - Delete Booth
+        [HttpDelete("{id}")]
+        public ActionResult<Booth> Delete(int id)
+        {
+            try
+            {
+                return _service.Delete(id);
```
Comment (asbamse) [inline BazarRestAPI/Controllers/BoothsController.cs]: Missing Ok()

**Comment by Applemelon** [comment]
Missing documentation in BoothRepository

File: infrastructure/BoothRepository.cs [hunk 4]
```diff
@@ -1,41 +1,60 @@
...
+            _ctx = ctx;
+        }
+        // Counts the amount of booths in database.
         public int Count()
         {
-            throw new NotImplementedException();
+            return _ctx.Booth.Count();
         }
-
+        // Creates Booth
```
Comment (JanvanZetten) [inline infrastructure/BoothRepository.cs]: if you write method comments with /// it is better

File: infrastructure/BoothRepository.cs [hunk 5]
```diff
@@ -1,41 +1,60 @@
...
         public Booth Create(Booth entity)
         {
-            throw new NotImplementedException();
+            // Making variable for when we have to hash password etc.
```
Comment (JanvanZetten) [inline infrastructure/BoothRepository.cs]: This comment does not make so much sense and you should not make things you do not need yet

**Review by JanvanZetten — CHANGES_REQUESTED** [review]

**Review by JanvanZetten — DISMISSED** [review]

File: infrastructure/BoothRepository.cs [hunk 6]
```diff
@@ -1,41 +1,82 @@
...
     public class BoothRepository : IRepository<Booth>
     {
+        private readonly BazarContext _ctx;
+
+        public BoothRepository(BazarContext ctx)
+        {
+            _ctx = ctx;
+        }
+        // Counts the amount of booths in database.
+        /// <summary>
```
Comment (JanvanZetten) [inline infrastructure/BoothRepository.cs:20]: Nice comments

**Review by JanvanZetten — APPROVED** [review]

**Pull request merged**